### ISSUE TITLE
Add a config parameter to set redis into ssl mode

### ISSUE THREAD

**Issue by s4g** [jvanasco/pyramid_session_redis#60]
I have a Redis server configured to work in ssl mode. 

A simple test of the server with redis.Redis(host=myHostname, port=6380, password=myPassword, ssl=True) works.

When I am trying to put sessions on the server I get 'Connection reset by peer' errors. 

There is no way I can find to communicate the 'ssl=True' parameter to the client.

Is it possible to add a config parameter to do that?

Thanks

**Comment by jvanasco** [comment]
Yes, but I'm not sure when I'll have time for this.  A PR against the current `main` (which is unreleased) would be welcome.

I think it could be done by:

1. RedisSessionFactory needs a `redis_ssl` parameter (https://github.com/jvanasco/pyramid_session_redis/blob/main/src/pyramid_session_redis/__init__.py#L114)
2. that needs to bubble down to populating `redis_options` with `ssl` and that value (https://github.com/jvanasco/pyramid_session_redis/blob/main/src/pyramid_session_redis/__init__.py#L529-L561)
3. I don't think any changes are needed to https://github.com/jvanasco/pyramid_session_redis/blob/main/src/pyramid_session_redis/connection.py#L61, as `**redis_options` should pass through.
4. We would need a new test , which is the hard part.  redis is spawned in github actions, so we may need to duplicate the action and figure out how to run the test suite "with ssl" and without.

In the meantime, you can immediately use the RedisSessionFactor `client_callable` as a hook to get this argument in there.  https://github.com/jvanasco/pyramid_session_redis/blob/main/src/pyramid_session_redis/__init__.py#L601-L606

`client_callable` will still pass in the config args, but you can ensure `ssl ` is put in there.    You should be able  to copy/paste the `get_default_connection` function into your own code, edit it to invoke redis with the options you want, and then submit that to the factory/initializer.
